Restructure SM2A documentation

ividito opened the pull request:
**Summary:** 

Reorganizing docs, consolidating README (based on MWAA-pruning branch and folder structure). Opening as a separate PR for easier reviews before merging to #352.

ividito committed:
Reorganize docs

stephenkilbourn reviewed (approved):
Wow, this was very informative. 👏

ividito commented on docs/contributing/airflow_variables.md:
```diff
@@ -0,0 +1,33 @@
+
+### Working with Airflow Variables
+
+Airflow variables allow passing secrets, configurations, etc., to tasks without embedding sensitive values in code.
```
Note - this doc will change with the addition of https://github.com/NASA-IMPACT/self-managed-apache-airflow/pull/19, but is more-or-less accurate as of right now

ividito commented on docs/operating/ingestion_configuration.md:
```diff
@@ -0,0 +1,63 @@
+# Ingestion Pipeline Overview
+
+This pipeline is designed to handle the ingestion of both vector and raster data. The ingestion can be performed using the `veda-discover` DAG. Below are examples of configurations for both vector and raster data.
```
This was lifted from the original README.md, a future PR will expand on this (including other DAGs and their behaviours)

smohiudd commented on docs/operating/ingestion_configuration.md:
```diff
@@ -0,0 +1,63 @@
+
+This pipeline is designed to handle the ingestion of both vector and raster data. The ingestion can be performed using the `veda-discover` DAG. Below are examples of configurations for both vector and raster data.
+
+## Ingestion Configuration
```
These configs occasionally change. Just so these docs are not out of sync can we instead point to the pipeline `template_dag_run_conf` (also need to makes sure they're correct).

botanical commented on README.md:
```diff
@@ -1,23 +1,19 @@
-- [infrastructure](./infrastructure/): Contains the terraform modules necessary to deploy all resources to AWS
-- [custom policies](./infrastructure/custom_policies/): Contains custom policies for the mwaa environment execution role
-- [scripts](./scripts/): Contains bash and python scripts useful for deploying and for running ingests
+- [dags](./dags/) contains the Directed Acyclic Graphs which constitute Airflow state machines. This includes the python for running each task as well as the python definitions of the structure of these DAGs. Files that define a DAG object are considered top-level DAG files, which are processed by Airflow.
```
```suggestion
- [dags](./dags/) contains the Directed Acyclic Graphs which constitute Airflow state machines. This includes the Python for running each task as well as the Python definitions of the structure of these DAGs. Files that define a DAG object are considered top-level DAG files, which are processed by Airflow.
```

botanical commented on docs/contributing/add_a_general_dag.md:
```diff
@@ -20,11 +46,9 @@
-- `ETL` - for ETL tasks outside of the Dynamic Vendor ETL Pipeline
-- `QAQC` - for data quality checks
-- `Template` - for templates
+- 
```
Is there anything to add here or can we delete this section?

botanical reviewed (commented):
Thanks for restructuring the docs! Just had one comment and question

ividito committed:
Update README.md

Co-authored-by: Jennifer Tran <12633533+botanical@users.noreply.github.com>

ividito committed:
Doc changes from review
- add tags we should be using
- direct readers to DAG run templates, as well as veda-data examples
